credsweeper/utils/util.py

```diff
@@ -73,8 +73,25 @@ def get_shannon_entropy(cls, data: str, iterator: Chars) -> float:
 
     @classmethod
     def read_patch_file(cls, path: str) -> List[str]:
-        with open(path, "r") as patch_file:
-            diff_data = patch_file.readlines()
+        """
+        Try to read the patch file using different encoding
+
+        Try to read the contents of the file according to the list of encodings "support_encoding" as soon as reading
+        occurs without any exceptions, the data is returned in the current encoding
+        """
+        supported_encoding = ["utf8", "utf16", "latin_1"]
+        diff_data = []
+        for encoding in supported_encoding:
+            try:
+                with open(path, "r", encoding=encoding) as patch_file:
+                    diff_data = patch_file.readlines()
+                break
+            except UnicodeDecodeError:
+                logging.warning(f"UnicodeDecodeError: Can't read patch content from \"{path}\" as {encoding}.")
+            except UnicodeError:
+                logging.warning(f"UnicodeError: Can't read patch content from \"{path}\" as {encoding}.")
+            except Exception as exc:
+                logging.error(f"Unexpected Error: Can't read \"{path}\" as {encoding}. Error message: {exc}")
```

Comment:
Please, explicitly process the case when failed to read contents with all provided encodings

Reply:
@Dmitriy-NK Why have you unresolved the issue?